### PR TITLE
common: add bundle test cases

### LINKED ISSUE #674: bundle: add test that compares calculated bundle hash against existing mainnet bundle
- we don't test against example bundles right now 
- we need a test that validates that computed normalised bundle hash is 13-free

### PR DESCRIPTION
**Pull request by bingyanglin**
Add more bundle test cases.
Fixes #674

### CODE REVIEW

**Review by thibault-martinez — APPROVED** [review]

File: consensus/bundle_validator/tests/bundle_validator.c [hunk 1]
```diff
@@ -132,6 +132,158 @@ void test_bundle_exceed_supply_neg_invalid() {
...
+void test_bundle_invalid_tx_with_wrong_last_idx() {
+  bundle_transactions_t *bundle;
+  bundle_status_t bundle_status = BUNDLE_NOT_INITIALIZED;
+  iota_transaction_t *txs[4];
+  bool exist = false;
+  tryte_t const *const trytes[4] = {TX_1_OF_4_VALUE_BUNDLE_TRYTES, TX_2_OF_4_VALUE_BUNDLE_TRYTES,
+                                    TX_3_OF_4_VALUE_BUNDLE_TRYTES, TX_4_OF_4_VALUE_BUNDLE_TRYTES};
+
+  transactions_deserialize(trytes, txs, 4, true);
+  transaction_set_current_index(txs[1], 2);
```
Comment (tsvisabo) [inline consensus/bundle_validator/tests/bundle_validator.c:191]: use `transaction_set_last_index`

File: consensus/bundle_validator/tests/bundle_validator.c [hunk 2]
```diff
@@ -132,6 +132,158 @@ void test_bundle_exceed_supply_neg_invalid() {
...
+  TEST_ASSERT(iota_consensus_bundle_validator_validate(&tangle, transaction_hash(txs[0]), bundle, &bundle_status) ==
+              RC_OK);
+  TEST_ASSERT(bundle_status == BUNDLE_INVALID_HASH);
+
+  bundle_transactions_free(&bundle);
+  transactions_free(txs, 4);
+}
+
+void test_bundle_invalid_hash_with_wrong_transaction_timestamp() {
+  bundle_transactions_t *bundle;
```
Comment (tsvisabo) [inline consensus/bundle_validator/tests/bundle_validator.c:264]: can you create an aux functions to gourp this logic?
something like:
`test_prepare_txs`

File: consensus/bundle_validator/tests/bundle_validator.c (hunk 1 above)
Comment (bingyanglin) [reply consensus/bundle_validator/tests/bundle_validator.c:191]: Thank you for pointing this out.
Fixs #1278

File: consensus/bundle_validator/tests/bundle_validator.c (hunk 2 above)
Comment (bingyanglin) [reply consensus/bundle_validator/tests/bundle_validator.c:264]: Yes, add aux function: test_prepare_4txs
Fixs #1278

**Pull request merged**